src/js/helpers/timeline/declarationHelper-groups.spec.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2020 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { extractGroupData, reduceSubgroups } from './declarationHelper';
+
+test('Group with parent information is correctly transform into group with subgroups information', () => {
+  const alreadyReducedData = { parent: [] };
+  const currentData = { group: 'child', subgroupOf: 'parent' };
+  const expectedData = { parent: ['child'], child: [] };
+
+  expect(reduceSubgroups(alreadyReducedData, currentData)).toStrictEqual(expectedData);
+});
+
+test('Parse group data as is without subgroups', () => {
+  const groupItems = [
+    { group: 'A' },
+    { group: 'B' },
+    { group: 'C' },
+  ];
+
+  const expectedGroupData = [
+    { name: 'A', subgroups: [] },
+    { name: 'B', subgroups: [] },
+    { name: 'C', subgroups: [] },
+  ];
+
+  expect(extractGroupData(groupItems)).toStrictEqual(expectedGroupData);
+});
+
+test('Flatten subgroups and keep ids in parent', () => {
+  const groupItems = [
+    { group: 'A' },
+    { group: 'B' },
+    { group: 'A-1', subgroupOf: 'A' },
```

Comment:
Does the order matter? What if the item with `subgroupOf: 'A'` appears before item with `group: 'A'`?

Reply:
It matters, this is the order in which the groups will be displayed in the timeline (first one on top)